src/types.ts

```diff
@@ -19,21 +19,21 @@ export interface UseThemeProps {
 
 export interface ThemeProviderProps {
   /** List of all available theme names */
-  themes?: string[]
+  themes: string[]
   /** Forced theme name for the current page */
   forcedTheme?: string
   /** Whether to switch between dark and light themes based on prefers-color-scheme */
-  enableSystem?: boolean
+  enableSystem: boolean
   /** Disable all CSS transitions when switching themes */
-  disableTransitionOnChange?: boolean
+  disableTransitionOnChange: boolean
   /** Whether to indicate to browsers which color scheme is used (dark or light) for built-in UI like inputs and buttons */
-  enableColorScheme?: boolean
+  enableColorScheme: boolean
   /** Key used to store theme setting in localStorage */
-  storageKey?: string
+  storageKey: string
   /** Default theme name (for v0.0.12 and lower the default was light). If `enableSystem` is false, the default theme is light */
-  defaultTheme?: string
+  defaultTheme: string
   /** HTML attribute modified based on the active theme. Accepts `class` and `data-*` (meaning any data attribute, `data-mode`, `data-color`, etc.) */
-  attribute?: string | 'class'
+  attribute: `data-${string}` | 'class'
```

Comment:
This is the only valid change here, the rest is wrong as those props are not required.

Reply:
it's up to you of course, but i would say that those types are indeed actually required.

you are [providing default values here](https://github.com/pacocoursey/next-themes/blob/main/src/index.tsx#L30-L36) for each of these props, but if any of those were to actually be `undefined` (which the current type definitions suggest is acceptable), it would lead to runtime errors. this is why i think it's more correct to make these types required.

note, this PR doesn't affect the implementation of the `<ThemeProvider>` component (you can see that this PR doesn't touch any of the example files, and that the examples both deployed correctly). 

in this PR, the type definitions which are actually not required (`value`, `nonce`, and `children`), have been left as optional